mac/build.sh

```diff
@@ -312,6 +116,30 @@ execBuildCommand() {
     fi
 }
 
+do_clean ( ) {
+  rm -rf "$KME4M_BASE_PATH/build"
+  rm -rf "$KM4MIM_BASE_PATH/build"
+  rm -rf "$KMTESTAPP_BASE_PATH/build"
+  rm -rf "$KEYMAN_ROOT/mac/setup/Install Keyman.app"
```

Comment:
So, fun discovery... the installer isn't output within the build folder.  I'd personally love to change that, but it _is_ a bit outside of scope here.

Caused me a bit of grief trying to figure out why I was getting a weird error from `/mac/setup/build.sh` about a PLIST entry already being defined.

Reply:
> in retrospect, maybe I should auto-clean it within the `setup/build.sh` script, even without top-level `clean`. We'd still run into it otherwise if locally calling without top-level `clean`.

I think that sounds sensible.

Reply:
Interestingly, it looks like that script should _already_ be cleaning it; I wonder why that was failing for me.

Sadly, I'm away from my Mac dev machine this week, so it'll be a little bit before I can investigate that.

Reply:
Found the culprit; we were using the wrong shell conditional check.

More details at https://github.com/keymanapp/keyman/pull/11444#discussion_r1615396979